presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -592,6 +593,52 @@ public Node visitResetSession(SqlBaseParser.ResetSessionContext context)
         return new ResetSession(getLocation(context), getQualifiedName(context.qualifiedName()));
     }
 
+    @Override
+    public Node visitGrant(SqlBaseParser.GrantContext context)
+    {
+        String grantee;
+        if (context.PUBLIC() != null) {
+            grantee = "PUBLIC";
+        }
+        else {
+            grantee = context.grantee.getText();
+        }
+
+        List<String> privileges;
+        if (context.ALL() != null && context.PRIVILEGES() != null) {
+            //List all privileges  declared in 'Privilege' enum in SPI
+            privileges = ImmutableList.of("SELECT", "DELETE", "INSERT");
```

Comment:
I am not sure I understand.
- Are you suggesting to somehow combine the two auto-generated methods, ALL() and PRIVILEGES(), into a single method and if so, may I ask, how? 
- I had added a test in testStatementBuilder.java as `printStatement("grant all privileges on foo to testAdmin");` that tests the "all privileges" case. Is this not sufficient? 

I have hard-coded values of privileges here, which I understand, is not a good choice. But I didn't see another way out.